### ISSUE TITLE
read_parquet column projection not working as expected

### ISSUE THREAD

**Issue by bnaul** [dask/dask#9104]
**What happened**:
Filtering out columns after calling `read_parquet` still causes the entire dataset to be loaded for downstream operations:
```
%%time
dd.read_parquet(
    "s3://nyc-tlc/trip data/fhv_tripdata_2015*.parquet",
)[[]].sum().compute()
CPU times: user 24 s, sys: 8.82 s, total: 32.8 s
Wall time: 1min 15s
Out[4]: Series([], dtype: float64)
```

**What you expected to happen**:
From my interpretation of #8692 we should get the same performance as
```
%%time
dd.read_parquet(
     "s3://nyc-tlc/trip data/fhv_tripdata_2015*.parquet",
      columns=[]
).sum().compute()
CPU times: user 147 ms, sys: 20.6 ms, total: 167 ms
Wall time: 706 ms
Out[5]: Series([], dtype: float64)
```

**Anything else we need to know?**:

**Environment**:

- Dask version: 2022.05.0
- Python version: 3.9.12
- PyArrow version: 8.0.0

Have not yet checked this behavior for any past dask versions.

cc: @rjzamora @martindurant @jcrist from #8692

**Comment by martindurant** [comment]
It looks like the "no columns" case isn't covered - but you get much faster performance if you include any one column. This probably also benefits a lot from the parquet pre-cache strategy.

**Comment by jcrist** [comment]
Thanks for filing this, I've pushed a fix at #9106.
